src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -81,28 +104,49 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
             raise CLIError('Please provide the unique logic name of an existing role')
         role_definition['name'] = matched[0].name
         # ensure correct logical name and guid name. For update we accept both
-        role_name = matched[0].role_name
+        role_name = _get_role_property(matched[0], 'role_name')
         role_id = matched[0].name
     else:
         role_id = _gen_guid()
 
     if not for_update and 'assignableScopes' not in role_definition:
         raise CLIError("please provide 'assignableScopes'")
 
+    Permission, RoleDefinition, RoleDefinitionProperties = get_sdk(cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                                                   'Permission', 'RoleDefinition', 'RoleDefinitionProperties', mod='models',
+                                                                   operation_group='role_definitions')
+    if not Permission:
```

Comment:
please remove this check as `permission` should always be available, even in older api-version